funtofem/mphys/mphys_meld.py

```diff
@@ -372,36 +537,89 @@ def __init__(
         self.beta = beta
         self.check_partials = check_partials
         self.linearized = linearized
+        self.body_tags = body_tags
+
+        if len(self.body_tags) > 0:  # make into lists, potentially for different bodies
+            if not hasattr(self.n, "__len__"):
+                self.n = [self.n] * len(self.body_tags)
+            if not hasattr(self.beta, "__len__"):
+                self.beta = [self.beta] * len(self.body_tags)
+            if not hasattr(self.linearized, "__len__"):
+                self.linearized = [self.linearized] * len(self.body_tags)
 
     def initialize(self, comm):
         self.nnodes_aero = self.aero_builder.get_number_of_nodes()
         self.nnodes_struct = self.struct_builder.get_number_of_nodes()
         self.ndof_struct = self.struct_builder.get_ndof()
 
-        if self.linearized:
-            self.meld = TransferScheme.pyLinearizedMELD(
-                comm, comm, 0, comm, 0, self.isym, self.n, self.beta
-            )
-        else:
-            self.meld = TransferScheme.pyMELD(
-                comm, comm, 0, comm, 0, self.isym, self.n, self.beta
-            )
+        self.bodies = []
+
+        if len(self.body_tags) > 0:  # body tags given
+            for i in range(len(self.body_tags)):
+                try:
+                    aero_gid = self.aero_builder.get_tagged_indices(
+                        self.body_tags[i]["aero"]
+                    )
+                except:
```

Comment:
Can you make these `excepts` more specific for the expected error? Basically `except NotImplementedError`